pkg/k8s/client_mock.go

```diff
@@ -91,82 +91,93 @@ func NewClientMock(stopCh <-chan struct{}, path string, smi bool) *ClientMock {
 	c.ServiceLister = c.informerFactory.Core().V1().Services().Lister()
 	c.EndpointsLister = c.informerFactory.Core().V1().Endpoints().Lister()
 	c.NamespaceLister = c.informerFactory.Core().V1().Namespaces().Lister()
+	c.TrafficSplitLister = c.splitInformerFactory.Split().V1alpha2().TrafficSplits().Lister()
 
 	// Start the informers.
-	c.informerFactory.Start(stopCh)
-
-	for t, ok := range c.informerFactory.WaitForCacheSync(stopCh) {
-		if !ok {
-			fmt.Printf("timed out waiting for controller caches to sync: %s", t.String())
-		}
-	}
+	c.startInformers(stopCh)
 
-	if smi {
+	if acl {
 		c.accessClient = fakeaccessclient.NewSimpleClientset(filterObjectsByKind(k8sObjects, AccessObjectKinds)...)
 		c.specsClient = fakespecsclient.NewSimpleClientset(filterObjectsByKind(k8sObjects, SpecsObjectKinds)...)
-		c.splitClient = fakesplitclient.NewSimpleClientset(filterObjectsByKind(k8sObjects, SplitObjectKinds)...)
 
 		c.accessInformerFactory = accessinformer.NewSharedInformerFactory(c.accessClient, 0)
 		c.specsInformerFactory = specsinformer.NewSharedInformerFactory(c.specsClient, 0)
-		c.splitInformerFactory = splitinformer.NewSharedInformerFactory(c.splitClient, 0)
 
 		trafficTargetInformer := c.accessInformerFactory.Access().V1alpha1().TrafficTargets().Informer()
 		httpRouteGroupInformer := c.specsInformerFactory.Specs().V1alpha1().HTTPRouteGroups().Informer()
 		tcpRouteInformer := c.specsInformerFactory.Specs().V1alpha1().TCPRoutes().Informer()
-		trafficSplitInformer := c.splitInformerFactory.Split().V1alpha2().TrafficSplits().Informer()
 
 		trafficTargetInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
 		httpRouteGroupInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
 		tcpRouteInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
-		trafficSplitInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
 
 		c.TrafficTargetLister = c.accessInformerFactory.Access().V1alpha1().TrafficTargets().Lister()
 		c.HTTPRouteGroupLister = c.specsInformerFactory.Specs().V1alpha1().HTTPRouteGroups().Lister()
 		c.TCPRouteLister = c.specsInformerFactory.Specs().V1alpha1().TCPRoutes().Lister()
-		c.TrafficSplitLister = c.splitInformerFactory.Split().V1alpha2().TrafficSplits().Lister()
 
 		// Start the informers.
-		c.accessInformerFactory.Start(stopCh)
-		c.specsInformerFactory.Start(stopCh)
-		c.splitInformerFactory.Start(stopCh)
-
-		for t, ok := range c.accessInformerFactory.WaitForCacheSync(stopCh) {
-			if !ok {
-				fmt.Printf("timed out waiting for controller caches to sync: %s", t.String())
-			}
+		c.startACLInformers(stopCh)
+	}
+
+	return c
+}
+
+// startInformers waits for the kubernetes core informers to start and sync.
+func (c *ClientMock) startInformers(stopCh <-chan struct{}) {
+	c.informerFactory.Start(stopCh)
+
+	for t, ok := range c.informerFactory.WaitForCacheSync(stopCh) {
+		if !ok {
+			fmt.Printf("timed out waiting for controller caches to sync: %s", t.String())
```

Comment:
Maybe not in this PR, but it could make sense to make the mock take a `t *testing.T` and to `t.Log` instead of `Printf`ing

Reply:
This is a great idea, I will open a new issue for this.

Reply:
https://github.com/containous/maesh/issues/578